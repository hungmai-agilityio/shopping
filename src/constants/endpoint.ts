export const END_POINT = {
  HOME: '/',
  SHOP: '/shop',
  ABOUT_US: '/about',
  CONTACT: '/contact',
  SIGN_IN: '/signIn',
  SIGN_UP: '/signUp',
  WISHLIST: '/wishlist',
  PRODUCTS: '/products',
  CATEGORY: '/category',
  COMMENT: '/comment',
  USERS: '/users',
  PROFILE: '/profiles',
  CART: '/cart',
  CHECKOUT: '/checkout'
};
