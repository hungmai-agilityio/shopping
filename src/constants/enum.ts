export enum TYPE {
  PRIMARY = 'primary',
  SECOND = 'second',
  THIRD = 'third',
  FILE = 'file',
  TEXT = 'text',
  PASSWORD = 'password',
  EMAIL = 'email',
  COMMENT = 'comment',
  CONTACT = 'contact'
}

export enum SIZE {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}

export enum TAG {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6',
  P = 'p',
  SPAN = 'span'
}

export enum FONT_SIZE {
  LARGE = 'text-heading',
  MEDIUM = 'text-4xl',
  SMALL = 'text-2xl',
  X_SMALL = 'lg:text-lg text-sm',
  MIMI = 'text-sm'
}

export enum FONT_WEIGHT {
  THIN = 'thin',
  LIGHT = 'light',
  NORMAL = 'normal',
  MEDIUM = 'medium',
  SEMIBOLD = 'semibold',
  BOLD = 'bold'
}

export enum CELL {
  TH = 'th',
  TD = 'td'
}

export enum CURRENT {
  USER = 'user',
  PROFILE = 'profile'
}

export enum GENDER {
  MALE = '0',
  FEMALE = '1',
  OTHER = '2'
}

export enum KEY {
  USER = 'currentUser'
}

export enum STATUS {
  SUCCESS = 'success',
  ERROR = 'error',
  WARNING = 'warning'
}

export enum TIME_STATUS {
  STALE_TIME = 1000 * 60 * 60, // milliseconds * seconds * minutes = 1 hour
  COOKIE_TIME = 60 * 60 * 24 * 2 // seconds * minutes * hour * day
}