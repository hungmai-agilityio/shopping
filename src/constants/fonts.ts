import { Mada, Poppins, Roboto, Signika } from 'next/font/google';

export const mada = Mada({
  weight: ['400', '700'],
  subsets: ['latin']
});

export const popping = Poppins({
  subsets: ['latin'],
  weight: ['400', '700']
})

export const roboto = Roboto({
  subsets: ['latin'],
  weight: ['400', '700']
})

export const signika = Signika({
  subsets: ['latin'],
  weight: ['400', '700']
})