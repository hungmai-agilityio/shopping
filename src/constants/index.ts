export * from '@/constants/fonts';
export * from '@/constants/enum';
export * from '@/constants/endpoint';
export * from '@/constants/messages';
export * from '@/constants/list';
