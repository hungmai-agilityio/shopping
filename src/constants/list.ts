import { END_POINT } from "@/constants/endpoint";

export const VALID_IMAGE = [
  'image/png',
  'image/jpeg',
  'image/jpg',
];

export const slideProducts = [
  {
    id: 'i7g6i9i2-8f89-6h15-9i0e-6j2f7j33g6i6',
    name: 'Basic T-Shirt',
    price: 15,
    description: 'A comfortable and stylish basic t-shirt.',
    category: 't-shirts',
    colors: ['White', 'Blue', 'Green'],
    sizes: ['S', 'M', 'L'],
    image: 'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
    imageDetail: [
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp',
      'https://i.ibb.co/njkCQHq/t-shirt-3.webp'
    ],
    rating: '10k',
    sell: 'Best quality',
    stoct: 1710
  },
  {
    id: 'u3k8h9f1-9g22-3h67-2g13-5k6g9k45g4t3',
    name: 'Sporty T-Shirt',
    price: 25,
    description: 'Perfect for your workout sessions.',
    category: 't-shirts',
    colors: ['Black', 'Gray', 'white', 'Maroon'],
    sizes: ['M', 'L', 'XL'],
    image: 'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
    imageDetail: [
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp',
      'https://i.ibb.co/WtgcmWR/t-shirt-2.webp'
    ],
    rating: '1k',
    sell: 'Best quality',
    stoct: 1118
  },
  {
    id: 'y4l7i8o1-2j33-4k56-8p92-7m9b8j45h7u8',
    name: 'Casual T-Shirt',
    price: 20,
    description: 'Stay casual and comfortable.',
    category: 't-shirts',
    colors: ['White', 'Yellow', 'Pink'],
    sizes: ['XS', 'S', 'M', 'L'],
    image: 'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
    imageDetail: [
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp',
      'https://i.ibb.co/J3GnxxJ/t-shirt-4.webp'
    ],
    rating: '2k',
    sell: 'Best quality',
    stoct: 1710
  },
  {
    id: 'o9b5l2e3-5t78-2v45-7h92-4c8h7d99a3u1',
    name: 'Graphic T-Shirt',
    price: 24,
    description: 'Unique graphic design for a standout look.',
    category: 't-shirts',
    colors: ['Blue', 'Purple'],
    sizes: ['S', 'M', 'L', 'XL'],
    image: 'https://i.ibb.co/fvhz0L9/t-shirt-1.webp',
    imageDetail: [
      'https://i.ibb.co/fvhz0L9/t-shirt-1.webp',
      'https://i.ibb.co/fvhz0L9/t-shirt-1.webp'
    ],
    rating: '9k',
    sell: 'Best quality',
    stoct: 1710
  }
];

export const slideImage = [
  'https://i.ibb.co/MRTgRYj/shirt-1.webp',
  'https://i.ibb.co/h1pRWGY/jacket-5-2.webp',
  'https://i.ibb.co/3d7b9SB/jean-2.webp'
];

export const navItems = [
  { name: 'Home', url: END_POINT.HOME },
  { name: 'About Us', url: END_POINT.ABOUT_US },
  { name: 'Contact', url: END_POINT.CONTACT }
];

export const itemsHome = [
  { name: 'Product', url: END_POINT.HOME },
  { name: 'Categories', url: END_POINT.HOME },
  { name: 'Login', url: END_POINT.SIGN_IN }
];

export const itemsShop = [
  { name: 'T-Shirt', url: END_POINT.HOME },
  { name: 'Jacket', url: END_POINT.HOME },
  { name: 'Shirt', url: END_POINT.HOME },
  { name: 'Jeans', url: END_POINT.HOME }
];

export const itemsCategories = [
  { name: 'Men', url: END_POINT.HOME },
  { name: 'Children', url: END_POINT.HOME },
  { name: 'Woman', url: END_POINT.HOME }
];

export const itemsContact = [{ name: 'mangcoding@gmail.com', url: '' }];