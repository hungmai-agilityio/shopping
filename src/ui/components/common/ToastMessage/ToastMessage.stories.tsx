import { Meta, StoryObj } from '@storybook/react';
import ToastMessage from '.';
import { STATUS } from '@/constants';

const meta: Meta<typeof ToastMessage> = {
  component: ToastMessage,
  tags: ['autodocs'],
  argTypes: {
    status: {
      description: 'Set status for toast'
    },
    message: {
      description: 'Message to show'
    }
  }
};

export default meta;

type Story = StoryObj<typeof ToastMessage>;

export const Success: Story = {
  args: {
    status: STATUS.SUCCESS,
    message: 'show message success'
  }
};

export const Error: Story = {
  args: {
    status: STATUS.ERROR,
    message: 'show message error'
  }
};

export const Warning: Story = {
  args: {
    status: STATUS.WARNING,
    message: 'show message warning'
  }
};
