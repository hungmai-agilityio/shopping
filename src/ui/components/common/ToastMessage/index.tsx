'use client';

import { clsx } from 'clsx';
import { useEffect, useState } from 'react';

// Constants
import { STATUS } from '@/constants';

interface ToastProps {
  status: STATUS.SUCCESS | STATUS.ERROR | STATUS.WARNING;
  message: string;
}

const ToastMessage = ({ status, message }: ToastProps) => {
  const [visible, setVisible] = useState<boolean>(true);

  const toastStyles = {
    [STATUS.SUCCESS]: 'bg-green-500',
    [STATUS.ERROR]: 'bg-red-500',
    [STATUS.WARNING]: 'bg-yellow-500'
  };

  useEffect(() => {
    const timer = setTimeout(() => {
      setVisible(false);
    }, 3000);

    return () => clearTimeout(timer);
  }, []);

  if (!visible) return null;

  return (
    <div
      className={clsx(
        'text-white font-semibold fixed top-4 right-4 py-2 px-4 rounded shadow-lg transition-opacity ease-in-out max-w-lg truncate',
        toastStyles[status]
      )}
    >
      {message}
    </div>
  );
};

export default ToastMessage;
