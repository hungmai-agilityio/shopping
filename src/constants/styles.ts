import { SIZE, STATUS, TYPE } from "@/constants/enum";

// Button
export const buttonSizeClasses = {
  [SIZE.SMALL]: 'w-12 h-12 text-xs',
  [SIZE.MEDIUM]: 'md:text-lg md:w-btn-md w-28 text-xs',
  [SIZE.LARGE]: 'w-full text-base'
};

export const buttonTypeClasses = {
  [TYPE.PRIMARY]: 'bg-gradient-primary text-white font-bold',
  [TYPE.SECOND]: 'bg-white text-purple-500 font-bold',
  [TYPE.THIRD]:
    'w-10 p-2 bg-gray-200 rounded-full drop-shadow-lg text-lg focus:bg-second focus:text-white focus:font-bold'
};

// Input
export const inputTypeClasses = {
  [TYPE.PRIMARY]: 'focus:border focus:border-green-300 h-12',
  [TYPE.SECOND]: 'border border-gray-400 h-12',
  [TYPE.THIRD]: `border border-gray-400 resize-none`
};

// Table
export const cellSizeClasses = {
  [SIZE.SMALL]: 'lg:w-tb-sm',
  [SIZE.MEDIUM]: 'lg:w-tb-md',
  [SIZE.LARGE]: 'lg:w-tb-xl'
};

// Toast
export const toastStyles = {
  [STATUS.SUCCESS]: 'bg-green-500',
  [STATUS.ERROR]: 'bg-red-500',
  [STATUS.WARNING]: 'bg-yellow-500'
};