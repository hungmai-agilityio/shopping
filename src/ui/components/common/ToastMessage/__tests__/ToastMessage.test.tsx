import { render, screen } from '@testing-library/react';

// Components
import ToastMessage from '@/ui/components/common/ToastMessage';
import { STATUS } from '@/constants';

describe('ToastMessage component', () => {
  const defaultProps = {
    status: STATUS.SUCCESS,
    message: 'Messages success'
  };

  it('Should be render with default props', () => {
    render(<ToastMessage {...defaultProps} />);
    const message = screen.getByText('Messages success');
    expect(message).toBeInTheDocument();
    expect(message).toHaveClass('bg-green-500');
  });

  it('should be render match to snapshot', () => {
    const { container } = render(<ToastMessage {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
