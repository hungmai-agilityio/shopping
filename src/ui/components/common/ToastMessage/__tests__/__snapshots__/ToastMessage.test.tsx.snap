// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage component should be render match to snapshot 1`] = `
<div>
  <div
    class="text-white font-semibold fixed top-4 right-4 py-2 px-4 rounded shadow-lg transition-opacity ease-in-out max-w-lg truncate bg-green-500"
  >
    Messages success
  </div>
</div>
`;
